src/lib/parquet/src/Flow/Parquet/BinaryReader/BinaryStreamReader.php

```diff
@@ -0,0 +1,299 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\Parquet\BinaryReader;
+
+use Flow\Parquet\BinaryReader;
+use Flow\Parquet\ByteOrder;
+use Flow\Parquet\DataSize;
+use Flow\Parquet\Exception\InvalidArgumentException;
+use Flow\Parquet\Exception\OutOfBoundsException;
+use Flow\Parquet\Exception\RuntimeException;
+
+final class BinaryStreamReader implements BinaryReader
+{
+    private int $bitPosition;
+
+    private int $fileLength;
+
+    public function __construct(private $handle, private readonly ByteOrder $byteOrder = ByteOrder::LITTLE_ENDIAN)
```

Comment:
As we store resource internally, we need `__destruct()` with close for the resource